### PR TITLE
LCAM-61-New endpoint for get old assessment

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-61)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/controller/MeansAssessmentController.java [hunk 1]
```diff
@@ -125,8 +125,7 @@ public ResponseEntity<ApiMeansAssessmentResponse> updateAssessment(@Parameter(de
                     schema = @Schema(implementation = ErrorDTO.class)
             )
     )
-    public ResponseEntity<ApiMeansAssessmentResponse> getOldAssessment(@PathVariable int financialAssessmentId,
-                                                                       @PathVariable String laaTransactionId) {
+    public ResponseEntity<ApiGetMeansAssessmentResponse> getOldAssessment(@PathVariable int financialAssessmentId, @PathVariable String laaTransactionId) {
```
Comment (Mhowell494) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/controller/MeansAssessmentController.java:128]: I don't think we should pass the LaaTransactionId as a path variable. This is meta information so shouldn't be part of the URL. Instead, this should be passed as a header.

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/controller/MeansAssessmentController.java (hunk 1 above)
Comment (loganathanMOJ) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/controller/MeansAssessmentController.java]: Fixed RequestHeader

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MeansAssessmentSectionSummaryBuilder.java [hunk 2]
```diff
@@ -19,18 +21,109 @@
...
+        if (StringUtils.isNotBlank(financialAssessmentDTO.getFassInitStatus())) {
+            assessmentStatus.setStatus(financialAssessmentDTO.getFassInitStatus());
+            assessmentStatus.setDescription(CurrentStatus.getFrom(financialAssessmentDTO.getFassInitStatus()).getDescription());
+            initialMeansAssessment.setAssessmentStatus(assessmentStatus);
+        }
+
+        ApiNewWorkReason newWorkReason = new ApiNewWorkReason();
+        if (StringUtils.isNotBlank(financialAssessmentDTO.getNewWorkReason())) {
+            newWorkReason.setCode(financialAssessmentDTO.getNewWorkReason());
+            newWorkReason.setDescription(NewWorkReason.getFrom(financialAssessmentDTO.getNewWorkReason()).getDescription());
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MeansAssessmentSectionSummaryBuilder.java:54]: Added NewWorkReason type

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MeansAssessmentSectionSummaryBuilder.java [hunk 3]
```diff
@@ -19,18 +21,109 @@
...
-    public List<AssessmentSectionSummaryDTO> buildAssessmentSectionSummary(List<AssessmentDTO> assessmentDTOList) {
 
-        List<AssessmentSectionSummaryDTO> assessmentSectionSummaryList = new ArrayList<>();
+    public void buildInitialAssessment(ApiGetMeansAssessmentResponse assessmentResponse, FinancialAssessmentDTO financialAssessmentDTO,
+                                       List<ApiAssessmentSectionSummary> assessmentSectionSummaryList,
+                                       Optional<AssessmentCriteriaEntity> criteriaEntity) {
+
+        ApiInitialMeansAssessment initialMeansAssessment = assessmentResponse.getInitialAssessment();
+        initialMeansAssessment.setId(financialAssessmentDTO.getId());
+        initialMeansAssessment.setCriteriaId(financialAssessmentDTO.getInitialAscrId());
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/builder/MeansAssessmentSectionSummaryBuilder.java]: Fixed now

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/MeansAssessmentService.java [hunk 4]
```diff
@@ -166,26 +166,30 @@ BigDecimal getDetailTotal(final ApiAssessmentDetail assessmentDetail, final bool
...
-    protected void mapIncomeEvidence(ApiMeansAssessmentResponse assessmentResponse, FinancialAssessmentDTO financialAssessmentDTO) {
+    protected void mapIncomeEvidence(ApiGetMeansAssessmentResponse apiGetMeansAssessmentResponse, FinancialAssessmentDTO financialAssessmentDTO) {
         List<FinAssIncomeEvidenceDTO> finAssIncomeEvidenceDTOList = financialAssessmentDTO.getFinAssIncomeEvidences();
 
+        ApiIncomeEvidenceSummary apiIncomeEvidenceSummary = new ApiIncomeEvidenceSummary();
+        apiIncomeEvidenceSummary.setEvidenceDueDate(financialAssessmentDTO.getIncomeEvidenceDueDate());
+        apiIncomeEvidenceSummary.setIncomeEvidenceNotes(financialAssessmentDTO.getIncomeEvidenceNotes());
+        apiIncomeEvidenceSummary.setUpliftAppliedDate(financialAssessmentDTO.getIncomeUpliftApplyDate());
+        apiIncomeEvidenceSummary.setUpliftRemovedDate(financialAssessmentDTO.getIncomeUpliftRemoveDate());
+        apiIncomeEvidenceSummary.setFirstReminderDate(financialAssessmentDTO.getFirstIncomeReminderDate());
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/MeansAssessmentService.java:190]: Added EvidenceReceivedDate

**Review by Mhowell494 — APPROVED** [review]

**Pull request merged**